### ISSUE TITLE
[Beats] Daily index is not created

### ISSUE THREAD

**Issue by kuisathaverat** [elastic/beats#11595]
We configured an index name to rotate the indices daily, but the index is only created with the date you started the service

For confirmed bugs, please report:
- Version: 7.0.0-rc1, 7.0.0-rc2
- Operating System: Linux on k8s
- Steps to Reproduce: 

* Starts a k8s cluster with Metricbeat and Packetbeat as daemonset, with the following output configuration

```
    output.elasticsearch:
      hosts: ['${ELASTICSEARCH_PROTO}://${ELASTICSEARCH_HOST:elasticsearch}:${ELASTICSEARCH_PORT:9200}']
      username: ${ELASTICSEARCH_USERNAME}
      password: ${ELASTICSEARCH_PASSWORD}
      index: "metricbeat-%{[agent.version]}-%{+yyyy.MM.dd}"
      ssl.enabled: true
      ssl.verification_mode: none
    xpack.monitoring.enabled: true
    setup.kibana.host: ${KIBANA_PROTO}://${KIBANA_HOST}:${KIBANA_PORT:5601}
    setup.kibana.username: ${KIBANA_USERNAME}
    setup.kibana.password: ${KIBANA_PASSWORD}
    setup.template.name: "metricbeat"
    setup.template.pattern: "metricbeat-*"
    setup.template.enabled: true
    setup.template.overwrite: true
```

```
    setup.template.overwrite: true
    xpack.monitoring.enabled: true
    queue.mem:
      events: 20000
    output.elasticsearch:
      hosts: ['${ELASTICSEARCH_PROTO}://${ELASTICSEARCH_HOST:elasticsearch}:${ELASTICSEARCH_PORT:9200}']
      username: ${ELASTICSEARCH_USERNAME}
      password: ${ELASTICSEARCH_PASSWORD}
      index: "packetbeat-%{[agent.version]}-%{+yyyy.MM.dd}"
      ssl.enabled: true
      ssl.verification_mode: none
      bulk_max_size: 2000
      worker: 2
      compression_level: 5
    xpack.monitoring.enabled: true
    setup.kibana.host: ${KIBANA_PROTO}://${KIBANA_HOST}:${KIBANA_PORT:5601}
    setup.kibana.username: ${KIBANA_USERNAME}
    setup.kibana.password: ${KIBANA_PASSWORD}
    setup.template.name: "packetbeat"
    setup.template.pattern: "packetbeat-*"
    setup.template.enabled: true
    setup.template.overwrite: true
```

`GET _cat/indices/*metric*`

```
green open metricbeat-7.0.0-rc1-2019.03.28-000001 LqeVmbWJSFKht2PkQVfi5g 1 1 48962310 0  80.4gb  40.2gb
```

`GET _cat/indices/*packet*`

```
green open packetbeat-7.0.0-rc1-2019.03.31-000003 g_9w21R1QDCUqW9X9IvSDQ 1 1 37772375 0 34.6gb  17.3gb
green open packetbeat-7.0.0-rc2-2019.04.02-000001 glKsCW6wTz6vSFAEcW6ofw 1 1  1181031 0  1.4gb 776.8mb
```

**Comment by MikaelElkiaer** [comment]
I just fell into the same trap, check out:
https://www.elastic.co/guide/en/beats/metricbeat/current/ilm.html
https://www.elastic.co/guide/en/beats/filebeat/current/ilm.html

**Comment by kuisathaverat** [comment]
I've checked the latest released version 7.0.0 and the latest snapshot 8.0.0-SNAPSHOT the issue is resolved, every time the ILM rollout an index if it is needed it creates a new index with the name based on the template configured.
